pilot/pkg/serviceregistry/kube/controller/fake.go

```diff
@@ -148,6 +150,18 @@ func (f *FakeController) ResyncEndpoints() error {
 	}
 	// endpoint processing may beat services
 	for _, ep := range eps {
+		// endpoint updates are skipped when the service is not there yet
+		if host, svc, ns := e.getServiceInfo(ep); host != "" {
+			_ = retry.UntilSuccess(func() error {
+				f.RLock()
+				defer f.RUnlock()
+				if f.servicesMap[host] == nil {
+					return fmt.Errorf("waiting for service %s in %s to be populated", svc, ns)
+				}
+				return nil
+			}, retry.Delay(time.Second), retry.Timeout(10*time.Second))
```

Comment:
Instead of waiting for service, why not just ignore these kinds of eps?

Reply:
I'm testing eds via TestMeshNetworks - making sure we get the gateway address for endpoints on a different network than the proxy. The endpoints are set manually as part of the test.